### ISSUE TITLE
Unable to use HAVING in SOQL query

### ISSUE THREAD

**Issue by cchrisv** [douglascayers-org/sfdx-mass-action-scheduler#105]
## Summary

I receive the error "Invalid SOQL Query
Expected "(", ",", ".", "FOR", "LIMIT", "OFFSET", "ORDER", or end of input but "H" found." when attempting to use HAVING in a SOQL query.

The query is 
SELECT Lead__c	, count(Id)
FROM Journey_Pipeline__c
GROUP BY Lead__c
HAVING count(Id) > 1

I am attempting to return leads that have a duplicate child record. This works fine in Developer console.

## Version of Mass Action Scheduler

2.2

## Steps To Reproduce

1. Create Mass Action Configuration
2. Select SOQL source type
3. Enter query 

SELECT Lead__c	, count(Id)
FROM Journey_Pipeline__c
GROUP BY Lead__c
HAVING count(Id) > 1

4. Select Validate Query

## Expected Behavior

Return leads with more than one child record.

## Actual Result

error "Invalid SOQL Query
Expected "(", ",", ".", "FOR", "LIMIT", "OFFSET", "ORDER", or end of input but "H" found."

## Source Type

- [ ] Reports
- [ ] List Views
- [X ] SOQL Queries

## Action Type

- [ X] Process Builder & Flows
- [ ] Workflow Rules
- [ ] Email Alerts
- [ ] Quick Actions
- [ ] Apex

## Does your configuration use Named Credentials?
Yes

**Comment by Xixiao007** [comment]
I can confirm this issue appears.

It applies to standard object too, for instance the SOQL below throws the same toast error message.

```
SELECT Name, COUNT(Id) FROM Account GROUP BY Name HAVING count(Id) > 1
```

**Comment by douglascayers** [comment]
Thanks for reporting this. Seems this is a limitation of the [soql-parse]() library -- doesn't seem to parse the `HAVING` clause in SOQL queries. I'll work on a fix for the upcoming 2.3 release.

**Comment by douglascayers** [comment]
To your observations, @Xixiao007,

> Afterwards, there seems yet another SOQL syntax validation, which fails, in client in the lightning component.
As a side note, it seems step 1 and 2 are done in the wrong order. IMO, it should be client validation first, Apex call second.

I agree, it needs to be refactored. Originally, I just wanted to ensure the query would run to give user's notice. Later I wanted a way to provide a better Field Mapping experience to let users choose fields from their queries without typing plain text, and so introduced the soql-parse library to get the details.

> Is the local SOQL syntax validation done in validateSoqlQueryAsync in MA_EditConfigCmpHelper.js? To be checked when I get time.

Yes

**Comment by douglascayers** [comment]
Release 2.3 is now available per #99, please see the [release notes](https://github.com/douglascayers-org/sfdx-mass-action-scheduler/wiki/Release-Notes-v2.3) for full details. Thank you!
